src/Cake.Core/CakeRuntime.cs

```diff
@@ -17,15 +17,62 @@ public sealed class CakeRuntime : ICakeRuntime
         public FrameworkName TargetFramework { get; private set; }
 
         /// <summary>
+        /// Gets a value indicating whether the current execution is running in debug mode.
+        /// </summary>
+        public bool IsDebug { get; private set; }
+
+        /// <summary>
         /// Initializes a new instance of the <see cref="CakeRuntime"/> class.
         /// </summary>
-        public CakeRuntime()
+        /// <param name="arguments">The arguments.</param>
+        public CakeRuntime(ICakeArguments arguments)
```

Comment:
This does not work as ICakeArguments is resolved much later on... One solution could be to wrap it around Lazy<T> or Func<T> which is both supported by Autofac for these "chicken-egg" issues..

Another way could be to extract some of the Core classes to a separate module which could be built after the Arguments module..

Thought? @gep13 @patriksvensson